ssl/statem/extensions_clnt.c

```diff
@@ -2015,3 +2015,107 @@ int tls_parse_stoc_psk(SSL_CONNECTION *s, PACKET *pkt,
 
     return 1;
 }
+
+EXT_RETURN tls_construct_ctos_client_cert_type(SSL_CONNECTION *sc, WPACKET *pkt,
+                                               unsigned int context,
+                                               X509 *x, size_t chainidx)
+{
+    sc->ext.client_cert_type_ctos = 0;
+    if (sc->client_cert_type != NULL) {
```

Comment:
I would tend to make this `if (sc->client_cert_type == NULL)` and return `EXT_RETURN_NOT_SENT` immediately here. Avoids subsequently additional indentiation

Reply:
Re "compressed RPK", I am not sure what RPK compression would mean? I don't believe there's a spec for that.